pinery-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java

```diff
@@ -0,0 +1,1031 @@
+      }
+    }
+    throw new IllegalArgumentException("ID '" + id + "' is not in expected format (e.g. SAM12, LIB345, or LDI78)");
+  }
+
+  @Override
+  public List<SampleProject> getSampleProjects() {
+    return template.query(queryAllSampleProjects, sampleProjectMapper);
+  }
+
+  @Override
+  public List<Sample> getSamples(Boolean archived, Set<String> projects, Set<String> types, ZonedDateTime before, ZonedDateTime after) {
+    List<Sample> samples = template.query(queryAllSamples, sampleMapper);
+    mapChildren(samples);
+    if (archived == null && (projects == null || projects.isEmpty()) && (types == null || types.isEmpty()) && before == null
+        && after == null) {
+      return samples;
+    } else {
+      return filterSamples(samples, archived, projects, types, before, after);
+    }
+  }
+
+  private List<Sample> filterSamples(List<Sample> unfiltered, Boolean archived, Set<String> projects, Set<String> types,
+      ZonedDateTime before,
+      ZonedDateTime after) {
+    Set<Filter<Sample>> filters = makeSampleFilters(archived, projects, types, before, after);
+    List<Sample> filteredSamples = new ArrayList<>();
+    for (Sample sample : unfiltered) {
+      boolean match = true;
+      for (Filter<Sample> filter : filters) {
+        if (!filter.matches(sample)) {
+          match = false;
+          break;
+        }
+      }
+      if (match) filteredSamples.add(sample);
+    }
+    return filteredSamples;
+  }
+
+  private Set<Filter<Sample>> makeSampleFilters(final Boolean archived, final Set<String> projects, final Set<String> types,
+      final ZonedDateTime before, final ZonedDateTime after) {
+    Set<Filter<Sample>> filters = new HashSet<>();
+
+    // archived filter
+    if (archived != null) {
+      filters.add(sam -> archived.equals(sam.getArchived()));
+    }
+
+    // projects filter
+    if (projects != null && !projects.isEmpty()) {
+      filters.add(sam -> projects.contains(sam.getProject()));
+    }
+
+    // types filter
+    if (types != null && !types.isEmpty()) {
+      filters.add(sam -> types.contains(sam.getSampleType()));
+    }
+
+    // before filter
+    if (before != null) {
+      filters.add(sam -> before.isAfter(toZonedDateTime(sam.getCreated())));
+    }
+
+    // after filter
+    if (after != null) {
+      filters.add(sam -> after.isBefore(toZonedDateTime(sam.getModified())));
+    }
+    return filters;
+  }
+
+  private static ZonedDateTime toZonedDateTime(Date from) {
+    return ZonedDateTime.ofInstant(from.toInstant(), ZoneId.systemDefault());
+  }
+
+  private interface Filter<T> {
+    public boolean matches(T object);
+  }
+
+  public List<Sample> getSamples() {
+    return getSamples(null, null, null, null, null);
+  }
+
+  private Sample addChildren(Sample parent) {
+    List<String> children = template.query(querySampleChildIdsBySampleId, new Object[] { parent.getId(), parent.getId() }, idListMapper);
+    if (children.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)